### PR TITLE
chore(clerk-js): Simplify dev browser handling

### PR DESCRIPTION
**Pull request by SokratisVidros**
## Description

1. Work only with URL-based session syncing dev instances and drop support for cookie-based syncing dev instances.
2. Align nomenclature across cookie, hearer, and search param names
3. Drop cookie handler in favor of dedicated cookie handlers, one for each cookie (__clerk_db_jwt, __session, __client_uat)

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

File: packages/clerk-js/src/ui/customizables/parseVariables.ts [hunk 1]
```diff
@@ -73,7 +73,7 @@ export const createFontSizeScale = (theme: Theme): Record<keyof typeof fontSizes
   return {
     xs: (numericValue * 0.6875).toString() + unit,
     sm: (numericValue * 0.75).toString() + unit,
-    md: (numericValue * 0.8125).toString() + unit,
+    md: (numericValue * 1).toString() + unit,
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/customizables/parseVariables.ts:76]: This should probably be reverted

**Review by nikosdouvlis — APPROVED** [review]
We need to apply this changes to AP as well, correct?

File: packages/clerk-js/src/core/devBrowser.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    // 1. Set network interceptors to Pass dev
+    fapiClient.onBeforeRequest(request => {
+      const devBrowserJWT = getDevBrowserJWT();
+      if (devBrowserJWT && request?.url) {
+        request.url = setDevBrowserJWTInURL(request.url, devBrowserJWT, true);
+      }
+    });
+
+    fapiClient.onAfterResponse((_, response) => {
+      const newDevBrowserJWT = response?.headers?.get('Clerk-Db-Jwt');
```
Comment (BRKalow) [inline packages/clerk-js/src/core/devBrowser.ts:56]: (nit) Can we make this a constant that lives with the others?

File: packages/clerk-js/src/core/devBrowser.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    });
+
+    // 1. If a cookie already exists, it might have SameSite=Strict. Re-set it to make sure it has SameSite=Lax
+    const existingDevBrowserCookie = getDevBrowserCookie();
+    if (existingDevBrowserCookie) {
+      removeDevBrowserCookie();
+      setDevBrowserCookie(existingDevBrowserCookie);
+    }
+
+    // 2. Get the JWT from hash or search parameters when the redirection comes from AP
```
Comment (BRKalow) [inline packages/clerk-js/src/core/devBrowser.ts:69]: Hedging considering we're going to drop the hash transport mechanism:

```suggestion
    // 2. Get the JWT from the URL when the redirection comes from AP
```

File: packages/clerk-js/src/utils/cookies/session.ts [hunk 4]
```diff
@@ -1,13 +1,27 @@
...
  *
- * For more information refer to the following document:
- *
- * https://docs.google.com/document/d/1F_vp7X4mhu0QXYG_89GVXnMoBlFZTjL33jMd4t120PM/edit#
  */
 export const sessionCookie = createCookieHandler(SESSION_COOKIE_NAME);
+
+export const removeSessionCookie = () => sessionCookie.remove();
+
+export const setSessionCookie = (token: string) => {
```
Comment (BRKalow) [inline packages/clerk-js/src/utils/cookies/session.ts:17]: I like this new pattern 👍

File: packages/clerk-js/src/core/devBrowser.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+  removeDevBrowserJWT(): void;
+}
+
+export type CreateDevBrowserOptions = {
+  frontendApi: string;
+  fapiClient: FapiClient;
+};
+
+export function createDevBrowser({ frontendApi, fapiClient }: CreateDevBrowserOptions): DevBrowser {
```
Comment (BRKalow) [inline packages/clerk-js/src/core/devBrowser.ts:25]: ❓ why not make this a class?

**Review by BRKalow — APPROVED** [review]
Looks good!

File: packages/clerk-js/src/core/devBrowser.ts (hunk 5 above)
Comment (SokratisVidros) [reply packages/clerk-js/src/core/devBrowser.ts:25]: Not a huge fan of classes 😆

File: packages/clerk-js/src/core/devBrowser.ts (hunk 3 above)
Comment (SokratisVidros) [reply packages/clerk-js/src/core/devBrowser.ts:69]: Good call. I have a to-do item for this, but I'd like us to see how V4 and V5 will work in AP before the final change. So we will get back to this next week.

File: packages/clerk-js/src/core/devBrowser.ts (hunk 2 above)
Comment (SokratisVidros) [reply packages/clerk-js/src/core/devBrowser.ts:56]: Fixed.

**Pull request merged**